packages/common/src/controllers/AccountController.ts

```diff
@@ -188,20 +166,27 @@ export default class AccountController {
   };
 
   login = async (email: string, password: string, referrer: string) => {
-    const { config, accessModel } = useConfigStore.getState();
-
     useAccountStore.setState({ loading: true });
 
-    const response = await this.accountService.login({ config, email, password, referrer });
+    try {
+      const response = await this.accountService.login({ email, password, referrer });
 
-    if (response) {
-      await this.afterLogin(response.user, response.customerConsents, accessModel);
+      if (!response) throw new Error("Couldn't login");
 
-      await this.favoritesController?.restoreFavorites();
-      await this.watchHistoryController?.restoreWatchHistory();
+      await this.afterLogin(response.user, response.customerConsents);
+      await this.favoritesController?.restoreFavorites().catch(logDev);
+      await this.watchHistoryController?.restoreWatchHistory().catch(logDev);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        if (error.message.toLowerCase().includes('invalid param email')) {
+          throw new FormValidationError({ email: [i18next.t('account:login.wrong_email')] });
+        } else {
+          throw new FormValidationError({ form: [i18next.t('account:login.wrong_combination')] });
+        }
+      }
     }
 
-    useAccountStore.setState({ loading: false });
+    return useAccountStore.setState({ loading: false });
```

Comment:
Does this need to be in a finally block so that we don't get stuck in 'loading'?

Reply:
See https://github.com/jwplayer/ott-web-app/pull/450/files/b700fbb7ea31cde67740bab99246a0b705058c55#r1484537050